### DISCUSSION TITLE
Enabling auto merge on a PR using the API

### DISCUSSION THREAD

**Discussion by dcoraboeuf** [community/community#24719] [Apps, API and Webhooks]
<p>Hi,</p>
<p>We’re very excited with the new Beta Auto merge feature for pull requests.</p>
<p>I did not see any way to enable auto merge on a pull request using the REST API at <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls</a> - is that already available?</p>
<p>Many thanks,<br>
Damien</p>

**Comment by vibou** [comment]
<p>Still no news regarding this. We are also interested in such API.</p>
<p>Just in case, another topic being duplicate of this one: <a href="https://github.community/t/enable-auto-merge-for-repository-via-api-v3/151331">enable-auto-merge-for-repository-via-api-v3</a></p>

**Comment by huehnerlady** [comment]
<blockquote>
<p>her topic being duplicate of this one: <a href="https://github.community/t/enable-auto-merge-for-repository-via-api-v3/151331">enable-auto-merge-for-repository-via-api-v3</a></p>
</blockquote>
<p>As far as I can see this one is about the flag in the pull request itself and the other one is about the setting in the repository. So to me they are not actually duplicates, but maybe I got that wrong?</p>

**Comment by huehnerlady** [comment]
<p>By now the auto-merge is also not in beta anymore, so I would have expected a flag in the API by now :thinking:</p>

**Comment by dcoraboeuf** [comment, marked as answer]
<p>Enabling auto-merge for a PR is now possible through the GraphQL API using the <code>enablePullRequestAutoMerge</code> mutation.</p>
<p>However, there doesn’t seem to be an API endpoint to enable this feature at repository level.</p>

**Reply by acaird** [reply]
This isn't the answer:
  1. it doesn't address the API question
  2. it doesn't address the "on a specific PR" question

**Reply by marverix** [reply]
<img width="809" height="500" alt="atbjrj" src="https://github.com/user-attachments/assets/346487c5-f234-471d-8aa8-39372350e06a" />

**Comment by hazcod** [comment]
<p>Is there a way to auto-merge PRs using this API that automatically pass all checks?<br>
e.g. always merge everything from dependabot if it works.</p>

**Comment by shayki5** [comment]
<p>How can I use it for auto-merge my PR via code?</p>

**Comment by pkazi** [comment]
`gh` cli has this support, this can be used by automation
https://cli.github.com/manual/gh_pr_merge

**Comment by testworksau** [comment]
Enabling / disabling the "Allow auto-merge" repository setting can now be set at the repository level via the REST API, using the `allow_auto_merge` parameter.

https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#update-a-repository

**Comment by 0xWDG** [comment]
Is this already possible trough the REST API?

**Reply by shamoon** [reply]
That setting allows auto-merge **at the repository level**, this thread is about enabling it on an individual PR via REST API (currently possible via cli or GraphQL API)

**Comment by navdotnetreqs** [comment]
So wait.. this is STILL not available? I want to create a PR via REST and have it auto-merge.
